service/history/execution/context.go

```diff
@@ -983,16 +983,19 @@ func (c *contextImpl) PersistStartWorkflowBatchEvents(
 			// TransactionID is set by shard context
 		},
 	)
-	return size, err
+	if err != nil {
+		return persistence.DataBlob{}, err
```

Comment:
this should be len(resp.Data) no?

Reply:
resp is most likely nil when an error is returned